### PR TITLE
LPS-152905 Add Preview for blogs

### PR DESCRIPTION
**Pull request by cgoncas**
### Motivation

As a content marketer, I want to view a preview of a blog image and web content in the info panel, so that I can more thoroughly understand the context of the asset at a glance.

### Proposed Solution

1. Add previewImage to the blogs info api: https://github.com/liferay-lima/liferay-portal/pull/3324
1. Read previewImage from the ContentDashboardAction
1. Remove downloadURL from the Preview Object.
 
### How to test it

1. Create a Blog
2. Add a cover image to that blog
3. Go to the content Dashboard
4. Check that the cover image is shown as the preview of the blog in the info panel of the blog.

### CODE REVIEW

**Comment by cgoncas** [comment]
Hey  @claraizquierdo !!

I need your help with this PR:

1. preview.url now can be null, so the link should not be shown when this URL is empty
2. What do you think about change the frontend/backend API and send the downloadURL in another field other than the preview?

Thanks!!

**Comment by claraizquierdo** [comment]
> Hey  @claraizquierdo !!
> 
> I need your help with this PR:
> 
> 1. preview.url now can be null, so the link should not be shown when this URL is empty
> 2. What do you think about change the frontend/backend API and send the downloadURL in another field other than the preview?
> 
> Thanks!!

Hi @cgoncas , the Preview component needs a few changes yes. I agree with you in both points 1 and 2. I'm going to start working on point 1 asap.

**Comment by claraizquierdo** [comment]
Hi @cgoncas, the two things that you pointed out in your comment are now fixed. Could you please review my last commit ([Return downloadURL in a separate field and not in the preview](https://github.com/liferay-tango/liferay-portal/pull/2654/commits/4769c200657c6f7eac6570eab7e8f98049ffcc04))? It touches 'a lot' of Java code and I will feel more confident if you review it :)

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java [hunk 1]
```diff
@@ -138,6 +138,8 @@ protected void doServeResource(
 					_toString(contentDashboardItem.getCreateDate())
 				).put(
 					"description", contentDashboardItem.getDescription(locale)
+				).put(
+					"downloadURL", contentDashboardItem.getDownloadURL()
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:142]: Hey @claraizquierdo !!

We don't add URLs to the API, we have already a download Action, could you take a look to the method _getFetchSharingButtonURL and obtain the Download URL the same way?

In the end you will need to mimic the _getDownloadURL method in the FilesEntryContentDashboardItem class:

```
private String _getDownloadURL() {
		return Optional.ofNullable(
			ServiceContextThreadLocal.getServiceContext()
		).map(
			ServiceContext::getLiferayPortletRequest
		).map(
			portletRequest -> {
				List<ContentDashboardItemAction> contentDashboardItemActions =
					getContentDashboardItemActions(
						_portal.getHttpServletRequest(portletRequest),
						ContentDashboardItemAction.Type.DOWNLOAD);

				if (!contentDashboardItemActions.isEmpty()) {
					ContentDashboardItemAction contentDashboardItemAction =
						contentDashboardItemActions.get(0);

					return contentDashboardItemAction.getURL();
				}

				return null;
			}
		).orElse(
			null
		);
	}
```

Thanks!!

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:142]: I can do it myself, but as you start to do the java code, if you want to continue, fell free.

Also, it is usually not  no mix changes in different modules.

We will usually create a commit for the changes in the API, another commit for the baseline, another commit for all the implementation, another commit with the changes in the content-dashboard-web.

Thanks!!

Comment (claraizquierdo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java:142]: Hi @cgoncas , thank you for the review! I will try to do it as you say :)

**Comment by claraizquierdo** [comment]
@cgoncas I have applied the changes you suggested. Let me know if everything is ok now :)

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @claraizquierdo !!

Could you please run SF in this PR. Don't forget to run it in every commit your do.

Thanks!!

**Comment by claraizquierdo** [comment]
ci:test:sf

**Review by claraizquierdo — APPROVED** [review]

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @claraizquierdo !!

Could you squash this commit: https://github.com/liferay-tango/liferay-portal/pull/2654/commits/2d76446866cbba63b2de50d0f21dd2f9d6b411b2 and https://github.com/liferay-tango/liferay-portal/pull/2654/commits/99684da826c65b22556780e673fe426baa547d91, i think have a extra SF commit is not very usefull, and if will make future rebases really hard to do.

Thanks!!

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:forward

**Pull request closed without merging**